client/signup/steps/site-type/index.jsx

```diff
@@ -81,6 +119,7 @@ export default connect(
 	state => ( {
 		siteType: getSiteType( state ) || 'blog',
 		hasInitializedSitesBackUrl: hasInitializedSites( state ) ? '/sites/' : false,
+		isJetpack: !! isJetpackSite( state, getSelectedSiteId( state ) ),
```

Comment:
As far as I know, signup step components are never used in the Jetpack connect flow, which has it's own top level components in `client/jetpack-connect`, so this check is not needed here.